File: 백준/Silver/1764. 듣보잡/듣보잡.cc
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <map>

using namespace std;

int main()
{
    // 입출력 속도 가속
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    // 듣도 못한을 저장할 맵
    map<string, int> deuddo;
    // 듣보잡을 저장할 벡터
    vector<string> deudbojab;

    int deuddoN, bodoM;

    cin >> deuddoN >> bodoM;

    for (int i = 0; i < deuddoN; i++) {
        string str;
        cin >> str;
        // 듣도 못한 저장
        deuddo.insert({ str, i });
    }

    for (int i = 0; i < bodoM; i++) {
        string str;
        cin >> str;

        // 보도 못한 사람이 듣도 못한 맵에도 있는 지 확인
        // 데이터를 끝까지 찾지 못했을 경우, iterator는 map.end()를 반환
        // 즉, 데이터가 맵에 있으면 듣보잡 벡터에 추가
        if (deuddo.find(str) != deuddo.end()) {
            deudbojab.push_back(str);
        }
    }

    sort(deudbojab.begin(), deudbojab.end());

    cout << deudbojab.size() << "\n";

    for (int i = 0; i < deudbojab.size(); i++) {
        cout << deudbojab[i] << "\n";
    }

    return 0;
}

File: 백준/Silver/1764. 듣보잡/듣보잡.java
import java.util.Scanner;
import java.util.Vector;
import java.util.Collections;
import java.util.HashMap;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int deuddoN, bodoM;

        deuddoN = sc.nextInt();
        bodoM = sc.nextInt();
        sc.nextLine();

        // 듣도 못한을 저장할 맵
        HashMap<String, Integer> deuddo = new HashMap<>(deuddoN);
        // 듣보잡을 저장할 벡터
        Vector deudbojab = new Vector<>();

        for (int i = 0; i < deuddoN; i++) {
            String str;
            str = sc.nextLine();
            // 듣도 못한 저장
            deuddo.put(str, i);
        }

        for (int i = 0; i < bodoM; i++) {
            String str;
            str = sc.nextLine();
            // 보도 못한 사람이 듣도 못한 맵에도 있는 지 확인
            // 데이터를 끝까지 찾지 못했을 경우, null를 반환
            // 데이터가 맵에 있으면 듣보잡 벡터에 추가
            if (deuddo.containsKey(str) == true) {
                deudbojab.add(str);
            }
        }

        Collections.sort(deudbojab);

        System.out.println(deudbojab.size());
        for (int i = 0; i < deudbojab.size(); i++) {
            System.out.println(deudbojab.get(i));
        }

        sc.close();
    }
}
